app/src/main/java/com/hfad/mymessenger/CreateMessageActivity.kt

```diff
@@ -17,35 +19,67 @@ class CreateMessageActivity : AppCompatActivity() {
         binding = ActivityCreateMessageBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        //TODO: Add a setOnClickListener to each Button using the binding
+        // Add a setOnClickListener to each Button using the binding
+
+        binding.sendExplicit.setOnClickListener {
+sendMessageToActivity()
+
+        }
+
+        binding.sendImplicit.setOnClickListener {
+sendMessageToOtherApp()
+        }
 
     }
 
     //Creates an Explicit Intent to start the ReceiveMessageActivity
     private fun sendMessageToActivity() {
-        //TODO: Create an explicit Intent for ReceiveMessageActivity
+        // Create an explicit Intent for ReceiveMessageActivity
+
+        val receiveMessageActivity = Intent (this, ReceiveMessageActivity:: class.java)
+
 
-        // TODO: use putExtra method to attach the string value from the message editText field
+        //  use putExtra method to attach the string value from the message editText field
         //      and the boolean "isChecked" value from the markUrgent checkbox (use the binding to get this)
 
+        receiveMessageActivity.putExtra(MESSAGE, binding.message.text.toString())
+        receiveMessageActivity.putExtra(URGENT, true)
+
+
+
+
+        //   Start the new Activity
+        startActivity(receiveMessageActivity)
 
-        //  TODO: Start the new Activity
 
     }
 
     //Creates an IMPLICIT intent, adds the EditText's message as a String, and sends it to a "Chooser" window for the user to pick
     fun sendMessageToOtherApp() {
         val dataIntent = Intent(Intent.ACTION_SEND)
-        dataIntent.type = "text/plain"
+        dataIntent.type = "string/text"
         //TODO: use putExtra method with the label "Intent.EXTRA_TEXT" to attach the user's message to the intent
         //  and send it to a "Chooser" window for the user to see it with the different apps
 
+        val title = resources.getString(R.string.choser_title)
```

Comment:
Your dataIntent is missing the users message. Use a putExtra, similar to the code in line 45, but using `dataIntent.putExtra`